launchers/registry_serve.go

```diff
@@ -41,11 +41,17 @@ func RegistryServe(ctx context.Context, engine *gin.Engine, egrp *errgroup.Group
 		return errors.Wrap(err, "Unable to initialize the namespace registry database")
 	}
 
-	registry.InitOptionsCache(ctx, egrp)
+	if param.Server_EnableUI.GetBool() {
+		registry.InitOptionsCache(ctx, egrp)
 
-	err = registry.InitCustomRegistrationFields()
-	if err != nil {
-		return err
+		err = registry.InitCustomRegistrationFields()
+		if err != nil {
+			return err
+		}
+
+		if err := registry.InitInstConfig(ctx, egrp); err != nil {
+			return err
+		}
```

Comment:
For consistency, can you make the err handling in lines 47-50 the same as 52-54?
```
	if err := registry.InitCustomRegistrationFields(); err != nil {
			return err
	}
```